File: example/src/main/java/com/example/collapsingtoolbarlayoutsubtitle/BaseToolbarLayout.java
package com.example.collapsingtoolbarlayoutsubtitle;

import android.support.annotation.Nullable;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
interface BaseToolbarLayout {

    void setTitle(@Nullable CharSequence text);

    void setSubtitle(@Nullable CharSequence text);

    void setExpandedTitleGravity(int gravity);

    void setCollapsedTitleGravity(int gravity);
}

File: example/src/main/java/com/example/collapsingtoolbarlayoutsubtitle/BaseToolbarLayout.kt
package com.example.collapsingtoolbarlayoutsubtitle

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
internal interface BaseToolbarLayout {

    fun setTitle(text: CharSequence?)

    fun setSubtitle(text: CharSequence?)

    fun setExpandedTitleGravity(gravity: Int)

    fun setCollapsedTitleGravity(gravity: Int)
}